### DISCUSSION TITLE
Script Validation error 관련 문의드립니다.

### DISCUSSION THREAD

**Discussion by JAESKAAA** [naver/ngrinder#963] [Q&A]
일전에 #942 통해서 나왔던 이슈와 동일한 증상 겪고있는데,
혹시 관련해서 업데이트된 내용이나 원인 파악 된 부분 있을지 알 수 있을까요 ?

실행환경 : docker
버전 : v.3.5.8

 아래 이슈내용 참고 부탁드립니다.

- Validate 버튼 클릭시 이슈 발생
- 응답은 아래와 같습니다.
- {
  "message" : "Handler dispatch failed; nested exception is java.lang.NoClassDefFoundError: Could not initialize class org.ngrinder.common.util.SystemInfoUtils",
  "success" : false,
  "stackTrace" : "org.springframework.web.util.NestedServletException: Handler dispatch failed; nested exception is java.lang.NoClassDefFoundError: Could not initialize class org.ngrinder.common.util.SystemInfoUtils\n\tat org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)\n\tat org.ngrinder.security.PluggablePreAuthFilter.doFilter(PluggablePreAuthFilter.java:52)\n\tat org.ngrinder.infra.servlet.PluggableServletFilter.doFilter(PluggableServletFilter.java:61)\n\tat org.ngrinder.security.PluggablePreAuthFilter.doFilter(PluggablePreAuthFilter.java:52)\n\tat org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1589)\n\tat org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)\n\tat org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)\nCaused by: java.lang.NoClassDefFoundError: Could not initialize class org.ngrinder.common.util.SystemInfoUtils\n\tat net.grinder.engine.agent.PropertyBuilder.buildJVMArgumentWithoutMemory(PropertyBuilder.java:208)\n\tat net.grinder.engine.agent.LocalScriptTestDriveService.doValidate(LocalScriptTestDriveService.java:131)\n\tat org.ngrinder.script.service.ScriptValidationService.validate(ScriptValidationService.java:117)\n\tat org.ngrinder.script.controller.FileEntryApiController.validate(FileEntryApiController.java:392)\n\t... 7 more\n"
}
<img width="1305" alt="image" src="https://user-images.githubusercontent.com/78363202/230834473-4f78a3b0-2d02-4b2c-8833-af9f2c7bc31e.png">

**Comment by zibersaioros** [comment]
저도 같은 증상입니다...
맥 M1에서 도커로 ngrinder 구동시 발생합니다.

**Reply by junoyoon** [reply]
@donggyu04 @imbyungjun 님. 이게 리눅스 tmp 삭제 이슈일까요?

**Reply by librekyu** [reply]
저도 같은 증상입니다 m1 docker, amazon-linux docker 에서도 동일하고
3.5.8, 3.5.7 모두 같은증상이 나왔습니다.

**Reply by junoyoon** [reply]
혹시 재시작 시에도 같은 증상이신가요?

**Reply by JAESKAAA** [reply]
네. 재시작도 해보고 이미지도 다시 풀받아서 해봐도 동일하더라구요.
#940 에서 3.5.5-p1버전으로 사용하면 괜찮다는 의견있어서 해당 버전으로 사용하고 있습니다.

**Reply by librekyu** [reply]
네 저도 재시작 해도 안됩니다

**Reply by donggyu04** [reply]
@junoyoon 도커 자체 이슈같은데요. 확인해보겠습니다.

**Reply by djkeh** [reply]
이거 해결이 되었나요?

**Reply by ghkdqhrbals** [reply]
MBP 인텔에서 도커 구동 후, 스크립트 체크 했을 때 같은 증상입니다.

사용중인 도커 버전은 아래와 같습니다.
```
Server: Docker Desktop 4.7.1 (77678)
 Engine:
  Version:          20.10.14
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.15
  Git commit:       87a90dc
  Built:            Thu Mar 24 01:46:14 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.5.11
  GitCommit:        3df54a852345ae127d1fa3092b95168e4a88e2f8
 runc:
  Version:          1.0.3
  GitCommit:        v1.0.3-0-gf46b6ba
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Reply by rlouol** [reply]
저도 사내에서 AKS에 배포하다가 똑같은 에러가 발생했었는데, ngrinder-core 내부에 oshi-core하고 net.java.dev.jna 버전을 각각 6.4.10 / 5.14.0 으로 올리고 JDK 11 이미지 베이스(adoptopenjdk/openjdk11:alpine-jre)로 Docker 이미지를 새로 만들어서 배포해보니까 스크립트 검증이 됐네요. 

내부 의존성들 버전 호환 확인이 필요해보입니다.
